src/main/java/com/synopsys/integration/alert/channel/jira/server/web/JiraServerCustomFunctionAction.java

```diff
@@ -85,7 +84,7 @@ public JiraServerCustomFunctionAction(AuthorizationManager authorizationManager,
                 }
                 createBadRequestIntegrationException(e);
             }
-            boolean jiraPluginInstalled = isJiraPluginInstalled(jiraAppService, password, username, JiraConstants.JIRA_APP_KEY);
+            boolean jiraPluginInstalled = JiraPluginCheckUtil.checkIsAppInstalledAndRetryIfNecessary(() -> jiraAppService.isAppInstalled(username, password, JiraConstants.JIRA_APP_KEY));
```

Comment:
This method is done almost exactly the same as the other of checkIsAppInstalledAndRetryIfNecessary. Can you change the method to take a BiFunction so that all you need here is to pass in username and password then a method reference to jiraAppService::isAppInstalled

Reply:
I would still need to create a lambda in order to pass the app key as the third param. I don't know if it's worth it at that point.

Reply:
Actually, both server and cloud use the same PluginManagerService. Could you just pass that, username, and password to the static method and run the call yourself?

Reply:
I could. What are you trying to solve here? You don't like the lambda?

Reply:
The lambda just feels a little unnecessary here. I think we could make it clearer with just normal method parameters. Lambdas to me generally mean different functionality, but we're always doing the same check.